kafka_utils/kafka_manual_throttle/main.py

```diff
@@ -0,0 +1,264 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import logging
+import sys
+
+import humanfriendly
+import six
+
+from kafka_utils.util import config
+from kafka_utils.util.zookeeper import ZK
+
+
+LEADER_THROTTLE_RATE_CONFIGURATION = "leader.replication.throttled.rate"
+FOLLOWER_THROTTLE_RATE_CONFIGURATION = "follower.replication.throttled.rate"
+
+
+def parse_opts():
+    parser = argparse.ArgumentParser(
+        description='Manually applies throttle values for brokers in a Kafka cluster',
+    )
+    parser.add_argument(
+        '--cluster-type',
+        '-t',
+        required=True,
+        help='cluster type, e.g. "standard"',
+    )
+    parser.add_argument(
+        '--cluster-name',
+        '-c',
+        help='cluster name, e.g. "uswest1-devc" (defaults to local cluster)',
+    )
+    parser.add_argument(
+        '--discovery-base-path',
+        dest='discovery_base_path',
+        type=str,
+        help='Path of the directory containing the <cluster_type>.yaml config',
+    )
+    parser.add_argument(
+        '--leader-throttle',
+        type=int,
+        required=False,
+        help='value (B/s) of the leader throttle to apply for brokers',
+    )
+    parser.add_argument(
+        '--follower-throttle',
+        type=int,
+        required=False,
+        help='value (B/s) of the leader throttle to apply for brokers',
+    )
+    parser.add_argument(
+        '--clear',
+        help='if set, throttles will be cleared instead of being set',
+        action='store_true',
+    )
+    parser.add_argument(
```

Comment:
wonder if we could have the interface be a little different, e,g:

```
kafka-throttle get ...
kafka-throttle set ...
```

I think this is a little more clear, but it's not necessary (since this works as well)

Reply:
Fair point, although few (none?) commands in kafka-utils follow this pattern, I'll keep it as is and get back to it later if we feel that using {{get/set}} is more natural 🙂